microfrontends/app-builder-menu/src/components/Menu.tsx

```diff
@@ -54,6 +52,12 @@ export function Menu(props: Props) {
     if (config) {
       request();
     }
+
+    if (menuOpen) {
+      window.addEventListener('beforeunload', (e) => {
+        sessionStorage.removeItem('menu_open');
+      });
```

Comment:
Why you need to wait `beforeunload` event for delete the session storage?

Reply:
Because deleting the session storage item on cleanup won't occur if the page gets refreshed.

Reply:
So I tried to make it directly remove the session storage item in `useEffect()` and tested it. Unfortunately, the menu doesn't render only once. That means that the item has been removed already before the subsequent render/s, which results in the menu not opening. So I think either we keep the `beforeunload` logic, or we could use `useRef` so `menuOpen` won't update on re-render. What do you think?